JDStatusBarNotification/JDStatusBarView.m

```diff
@@ -58,17 +58,45 @@ - (void)layoutSubviews;
 
   // label
   CGFloat topLayoutMargin = JDStatusBarRootVCLayoutMargin().top;
+    
+    CGFloat labelY = self.textVerticalPositionAdjustment + topLayoutMargin + 1;
+    CGFloat height = self.bounds.size.height - topLayoutMargin - 1;
+    
+    // adjust for iPhone X
+    if (topLayoutMargin > 0){
+		switch (_iphoneXSize) {
+			case JDStatusBarIphoneXSizeMini:
+				labelY = self.textVerticalPositionAdjustment + topLayoutMargin - 8.0 - 3.0; // 3.0 to adjust correctly the label
+				height = self.bounds.size.height - topLayoutMargin + 8.0;
+				break;
+			default:
+				break;
+		}
+    }
+    
   self.textLabel.frame = CGRectMake(0,
-                                    self.textVerticalPositionAdjustment + topLayoutMargin + 1,
+                                    labelY,
                                     self.bounds.size.width,
-                                    self.bounds.size.height - topLayoutMargin - 1);
+                                    height);
 
   // activity indicator
   if (_activityIndicatorView ) {
     CGSize textSize = [self currentTextSize];
     CGRect indicatorFrame = _activityIndicatorView.frame;
-    indicatorFrame.origin.x = round((self.bounds.size.width - textSize.width)/2.0) - indicatorFrame.size.width - 8.0;
-    indicatorFrame.origin.y = ceil(1+(self.bounds.size.height - indicatorFrame.size.height + topLayoutMargin)/2.0);
```

Comment:
I think we should be able to keep the original logic in line 70 & 71 here, if we use `textLabel.bounds` instead of `self.bounds`. Then we don't need the new custom logic. Could you check, if that would work?

Reply:
Done and it's still work